evap/staff/views.py

```diff
@@ -411,9 +411,9 @@ def semester_raw_export(request, semester_id):
         _('#Participants'), _('#Comments'), _('Average grade')])
     for course in semester.course_set.all():
         degrees = ", ".join([degree.name for degree in course.degrees.all()])
-        course.avg_grade, course.avg_deviation = calculate_average_grades_and_deviation(course)
-        if course.state in ['evaluated', 'reviewed', 'published'] and course.avg_grade is not None:
-            avg_grade = "{:.1f}".format(course.avg_grade)
+        distribution = calculate_average_distribution(course)
+        if course.state in ['evaluated', 'reviewed', 'published'] and distribution is not None:
```

Comment:
Sometimes we explicitly compare calculated values to None, sometimes we just say `if distribution` or `if course.avg_grade`. I think using `if a is not None` is the better way since it makes the intent more obvious (and also documents that None is a valid value here). I had a really hard time understanding why we use None as values for counts and distributions all over the results calculation and storing and think this might have helped. What do you guys think?

Reply:
about understanding the Nones, it really just means "nothing to see here". we could also use tuples with all zeroes, but then we would still need to check whether the sum is zero e.g. to avoid divisions by zero when averaging, and before displaying. in case of the grade, using zero as "nothing to see here" is even weirder in my opinion.

besides being more explicit about the None, do you have any suggestions?

Reply:
I agree that setting the value to None instead of to 0 is the better choice. I didnt't have any better idea.